packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -27,6 +27,14 @@ export interface GridFilterFormProps {
   applyFilterChanges: (item: GridFilterItem) => void;
   applyMultiFilterOperatorChanges: (operator: GridLinkOperator) => void;
   deleteFilter: (item: GridFilterItem) => void;
+  hasLinkOperatorColumn?: boolean;
+  linkOperators?: (GridLinkOperator.And | GridLinkOperator.Or)[];
+  columnsSort?: 'asc' | 'desc';
+  deleteIconContainerSx?: SxProps<Theme>;
```

Comment:
@mnajdova in the core how to you handle when you want the user to easily customize the style of nested elements with `sx` ?

Do you pass down the `sx` function like this or do you expose the classes and let the user nest inside the root sx ?

Reply:
I would recommend using `components` and `componentsProps` for this. Then developers can do:

```
<GridFilterForm
  componentsProps={{ deleteIconContainer: { sx: /* ... */ } }}
  ...
/>
```

At least this is what we do in the `@mui/base`.

Reply:
Do you mean using `components` props to let dev use a custom filter like that?

```js
const CustomFilterForm = (props) => <GridFilterForm
	componentsProps={{ deleteIconContainer: { sx: /* ... */ } }}
	{...props}
/>

<DataGrid
	...
	components={{
		filterForm: CustomFilterForm
	}}
```

For now, I tried to use nesting classes as follows. Considering that `<FilterPanel />` is the only component user can have access. The other components are considered as nested ones, and can only be customized by CSS selectors.

```js
<DataGrid
	...
	componentsProps={{
		filterPanel: {
			sx: {
				"& .MuiDataGrid-filterForm": { ... },
				"& .MuiDataGrid-closeIconController": { ... },
				...
			}
		}
	}}
```

Reply:
Sorry, for some reason I have missed this comment. If the other nested components can only be customized by CSS selectors, why are you adding this new prop `deleteIconContainerSx`. It contradicts with the previous statement.

My point was that, we don't add first class props on the component for prop that should be passed on some nested elements, we prefer using `componentsProps` for this use-case.